service/history/api/startworkflow/api.go

```diff
@@ -398,11 +391,12 @@ func (s *Starter) resolveDuplicateWorkflowID(
 		currentWorkflowConditionFailed.RunID,
 	)
 
+	newRunID := uuid.NewString()
```

Comment:
This is a notable change. Without it, MultiOps gets into a deadlock since the lock for this RunID is already obtained at the start.

Reply:
There are some designs that depend on the capability on starting a specific runID, so I don't want to close the door for that. But we can do refactoring to make that happen later. I actually don't see why it's necessary to create the same workflow again below which causes the deadlock problem.

Plz add your note as comments in the code.